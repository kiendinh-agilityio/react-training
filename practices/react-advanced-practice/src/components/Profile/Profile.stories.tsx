import Profile from '.';

export default {
  title: 'Profile',
  component: Profile,
  parameters: {
    layout: 'centered',
  },
};

export const ProfileUser = {
  args: {
    fullName: 'Esthera Jackson',
    email: 'esthera@simmmple.com',
    avatarUrl: 'https://phimtat.vn/wp-content/uploads/2023/11/avt-3d.jpg',
  },
};
