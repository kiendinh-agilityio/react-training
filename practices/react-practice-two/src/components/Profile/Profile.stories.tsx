import Profile from '.';

// import images
import { SerhiyHipskyy } from '@/assets/images';

export default {
  title: 'Profile',
  component: Profile,
  parameters: {
    layout: 'centered',
  },
};

export const ProfileUser = {
  args: {
    avatarUrl: SerhiyHipskyy,
    fullName: 'Serhiy Hipskyy',
    position: 'CEO Universal',
  },
};

export const DefaultAvatar = {
  args: {
    fullName: 'Justus Menke',
    position: 'CEO Eronaman',
  },
};
