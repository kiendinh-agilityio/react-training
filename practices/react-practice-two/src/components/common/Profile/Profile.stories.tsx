import Profile from '.';

export default {
  title: 'Profile',
  component: Profile,
  parameters: {
    layout: 'centered',
  },
};

export const ProfileUser = {
  args: {
    avatarUrl: 'images/SerhiyHipskyy.png',
    fullName: 'Serhiy Hipskyy',
    position: 'CEO Universal',
  },
};
