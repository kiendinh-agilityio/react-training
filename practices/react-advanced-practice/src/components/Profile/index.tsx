import { Box, Flex } from '@radix-ui/themes';

// Import components
import { Avatar, Text } from '@/components/common';

interface ProfileProps {
  fullName: string;
  email: string;
  avatarUrl: string;
}

const Profile = ({ fullName, email, avatarUrl }: ProfileProps) => (
  <Box className="flex gap-[15px] text-left">
    <Avatar fallback src={avatarUrl} />
    <Flex direction="column">
      <Text weight="bold" content={fullName} className="font-bold leading-base" />
      <Text content={email} className="text-gray" />
    </Flex>
  </Box>
);

export default Profile;
