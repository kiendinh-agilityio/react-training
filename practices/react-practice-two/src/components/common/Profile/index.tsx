// import common components
import { Avatar, Paragraph } from '@/components/common';

// import types
import { ParagraphVariant, TextSize } from '@/types';

interface ProfileProps {
  fullName: string;
  position: string;
  avatarUrl: string;
}

const Profile = ({ fullName, position, avatarUrl }: ProfileProps) => (
  <div className="flex items-center gap-5">
    <Avatar url={avatarUrl} alt={fullName} />
    <div>
      <Paragraph variant={ParagraphVariant.Medium} text={fullName} className="leading-base mb-1" />
      <Paragraph size={TextSize.ExtraSmall} text={position} />
    </div>
  </div>
);

export default Profile;
