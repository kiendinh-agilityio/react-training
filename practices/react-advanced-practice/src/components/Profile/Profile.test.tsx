import { render } from '@testing-library/react';
import Profile from '.';

describe('Profile Component', () => {
  const mockProps = {
    fullName: 'Esthera Jackson',
    email: 'esthera@simmmple.com',
    avatarUrl: 'https://phimtat.vn/wp-content/uploads/2023/11/avt-3d.jpg',
  };

  it('renders correctly with valid props', () => {
    const { container, getByText } = render(<Profile {...mockProps} />);

    expect(getByText('Esthera Jackson')).toBeInTheDocument();
    expect(getByText('esthera@simmmple.com')).toBeInTheDocument();

    // Snapshot test
    expect(container).toMatchSnapshot();
  });
});
