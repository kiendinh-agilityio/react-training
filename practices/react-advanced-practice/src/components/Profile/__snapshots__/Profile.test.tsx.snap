// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Component renders correctly with valid props 1`] = `
<div>
  <div
    class="rt-Box flex gap-[15px] text-left"
  >
    <span
      class="rt-reset rt-AvatarRoot rt-r-size-3 rt-variant-soft rounded-base w-[40px] h-[40px] "
      data-accent-color=""
    >
      <span
        class="rt-AvatarFallback"
      />
    </span>
    <div
      class="rt-Flex rt-r-fd-column"
    >
      <p
        class="rt-Text rt-r-weight-bold text-sm font-bold leading-base"
      >
        Esthera Jackson
      </p>
      <p
        class="rt-Text rt-r-weight-regular text-sm text-gray"
      >
        esthera@simmmple.com
      </p>
    </div>
  </div>
</div>
`;
